### PR TITLE
Remove the test skips for progresses API (TDD approach)

### PR DESCRIPTION
**Pull request by heyrandhir**
This pull request follows a Test-Driven Development (TDD) approach to remove the test skips for the progresses API. Additionally, this PR introduces the use of the INTERNAL_SERVER_ERROR_MESSAGE constant. This constant allows for consistent error handling by providing appropriate 500 error responses with relevant messages, while avoiding the exposure of exact error details.

The Parent ticket can be found [here](https://github.com/Real-Dev-Squad/website-backend/issues/1076)

### CODE REVIEW

File: controllers/progresses.js [hunk 1]
```diff
@@ -112,8 +113,9 @@ const getProgress = async (req, res) => {
         message: error.message,
       });
     }
-    return res.status(400).json({
-      message: error.message,
+    logger.error(error.message);
+    return res.status(500).json({
+      message: INTERNAL_SERVER_ERROR_MESSAGE,
     });
   }
```
Comment (Pratiyushkumar) [inline controllers/progresses.js:120]: why have we written  this twice
logger.error(error.message);
    return res.status(500).json({
      message: INTERNAL_SERVER_ERROR_MESSAGE,
    });
  }

**Review by Pratiyushkumar — COMMENTED** [review]
rest everything LGTM

File: controllers/progresses.js (hunk 1 above)
Comment (heyrandhir) [reply controllers/progresses.js:120]: logger will log it in our logs. while the `res.status(500).json({`will send the reponse.

**Review by sahsisunny — APPROVED** [review]

File: controllers/progresses.js (hunk 1 above)
Comment (Pratiyushkumar) [reply controllers/progresses.js:120]: you this answer gives me the meaning of the logger,
my doubt was we have mentioned twice the  line in the code why was that needed

Comment (heyrandhir) [reply controllers/progresses.js:120]: I am unable to get you. which line has been mentioned twice. can you tell the line no's which you think are repeated.

Comment (heyrandhir) [reply controllers/progresses.js:120]: As discussed in the call @Pratiyushkumar  I am resolving this.

**Pull request merged**